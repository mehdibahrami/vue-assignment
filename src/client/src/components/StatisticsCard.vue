<template>
  <Card :title="title" :hasPadding="hasPadding">
    <!-- Statistic -->
    <div class="statistic">
      {{ statisticValue }}
      <!-- Statistic unit -->
      <span v-if="unit" class="statistic-unit">{{ unit }}</span>
    </div>
  </Card>
</template>

<script>
import { Card } from "@/components";

export default {
  name: "StatisticsCard",
  components: {
    Card,
  },
  props: {
    title: {
      type: String,
    },
    unit: {
      type: String,
    },
    statistic: {
      type: [Number, String],
      required: true,
    },
    hasPadding: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    statisticValue: function () {
      // Check the statistic prop if it is not null pass the value otherwise pass -
      return this.statistic ? this.statistic : "-";
    },
  },
};
</script>

<style scoped lang="scss">
.statistic {
  display: flex;
  align-items: baseline;
  font-size: $font-size-2xl;
  line-height: 2rem;
  font-weight: $font-weight-semibold;
  color: $secondary;
}

.statistic-unit {
  margin-left: 0.25rem;
  font-size: $font-size-sm;
  line-height: 1.25rem;
  font-weight: $font-weight-medium;
  color: $gray;
}
</style>
