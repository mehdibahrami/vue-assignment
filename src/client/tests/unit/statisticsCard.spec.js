import { shallowMount } from "@vue/test-utils";
import StatisticsCard from "@/components/cards/StatisticsCard.vue";

describe("StatisticsCard", () => {
  let wrapper;

  it("renders correctly", () => {
    // shallow mounting the main component
    // Stub card component replacing with its template to load slotted component
    wrapper = shallowMount(StatisticsCard, {
      global: {
        stubs: {
          Card: {
            template: '<div class="card-body"><slot></slot></div>',
          },
        },
      },
      props: {
        title: "Energy",
        statistic: 53.2,
        unit: "kW",
      },
    });

    expect(wrapper.find("div.statistic").text()).toBe("53.2kW");
    expect(wrapper.element).toMatchSnapshot();
  });
});
