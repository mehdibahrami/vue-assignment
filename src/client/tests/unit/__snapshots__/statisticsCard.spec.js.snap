// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsCard renders correctly 1`] = `
<div
  class="card-body"
  haspadding="true"
  title="Energy"
>
  
  <!-- Statistic -->
  <div
    class="statistic"
  >
    53.2
    <!-- Statistic unit -->
    <span
      class="statistic-unit"
    >
      kW
    </span>
  </div>
  
</div>
`;
